docs/integration-docs-guide.md

```diff
@@ -0,0 +1,273 @@
+  before the sample message screenshot.
+  ```
+  {!congrats.md!}
+  ```
+
+* **Example rendering:**
+  ```
+  **Congratulations! You're done!**
+
+  Your WebhookName notifications may look like:
+  ```
+
+### `{!download-python-bindings.md!}` macro
+
+* **About:** Links to Zulip's [API page](https://zulipchat.com/api/) to download
+  and install Zulip's API bindings.
+
+* **Contents:** See [source][9].
+
+* **Example usage:** Currently mostly used in non-webhook integrations docs
+  under `templates/zerver/integrations/<integration_name>.md`.
+  ```
+  {!download-python-bindings.md!}
+  ```
+
+* **Example rendering:**
+  ```
+  Download and install our [Python bindings and example scripts](/api)
+  on the server where the IntegrationName bot will live.
+  ```
+
+### `{!change-zulip-config-file.md!}` macro
+
+* **About:** Instructs users to create a bot and specify said bot's
+  credentials in the config file for a given non-webhook integration.
+
+* **Contents:** See [source][10].
+
+* **Example usage:** Usually used in non-webhook integration docs under
+  `templates/zerver/integrations/<integration_name>.md`.
+  ```
+  {!change-zulip-config-file.md!}
+  ```
+
+* **Example rendering:**
+  ```
+  On your Zulip settings page, create a bot for Codebase.
+
+  Next, open `integrations/codebase/zulip_codebase_config.py` with your
+  favorite editor, and change the following lines to specify the email
+  address and API key for your Codebase bot:
+
+      ZULIP_USER = "codebase-bot@example.com"
+      ZULIP_API_KEY = "0123456789abcdef0123456789abcdef"
+      ZULIP_SITE = "http://localhost:9991/api"
+  ```
+
+### `{!git-append-branches.md!}` and `{!git-webhook-url-with-branches.md!}`
+
+* **About:** These two macros explain how to specify a list of branches
+  in the webhook URL to filter notifications in our Git-related webhooks.
+
+* **Contents:** See [git-append-branches][12] and
+  [git-webhook-url-with-branches][13].
+
+* **Example usage:** Used exclusively in Git integrations.
+  ```
+  {!git-append-branches.md!}
+  {!git-webhook-url-with-branches.md!}
+  ```
+
+### Other useful macros
+
+* `{!webhook-url.md!}` - Used internally by `{!create-bot-construct-url.md!}`
+  to generate the webhook URL. See [source][11].
+
+* `{!zulip-config.md!}` - Used internally by `{!change-zulip-config-file.md!}`
+  to specify the lines in the config file for a non-webhook integration.
+  See [source][15].
+
+* `{!webhook-url-with-bot-email.md!}` - Used in certain non-webhook integrations
+  to generate URLs of the form (see [source][14]):
+  ```
+  https://bot_email:bot_api_key@yourZulipDomain.zulipchat.com/api/v1/external/beanstalk
+  ```
+  A good example is
+  [Zulip's Beanstalk integration](https://zulipchat.com/integrations/#beanstalk)
+
+[1]: https://github.com/zulip/zulip/blob/master/templates/zerver/help/include/create-stream.md
+[2]: https://github.com/zulip/zulip/blob/master/zerver/lib/integrations.py#L51
```

Comment:
You'll need to change these links to go off a specific commit, so that the line numbers don't break as `master` changes.